### PR TITLE
feat(files): Expose chunked upload config via capabilities

### PR DESCRIPTION
**Pull request by provokateurin**
## Summary

Part of https://github.com/nextcloud/server/issues/47682.
https://github.com/nextcloud-libraries/nextcloud-upload/pull/1444

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files/lib/App.php [hunk 1]
```diff
@@ -44,9 +45,9 @@ public static function getNavigationManager(): INavigationManager {
 	public static function extendJsConfig($settings): void {
 		$appConfig = json_decode($settings['array']['oc_appconfig'], true);
 
-		$maxChunkSize = (int)Server::get(IConfig::class)->getAppValue('files', 'max_chunk_size', (string)(100 * 1024 * 1024));
 		$appConfig['files'] = [
-			'max_chunk_size' => $maxChunkSize
+			'max_chunk_size' => ChunkedUploadConfig::getMaxChunkSize(),
+			'max_parallel_count' => ChunkedUploadConfig::getMaxParallelCount(),
 		];
```
Comment (susnux) [inline apps/files/lib/App.php:51]: Does not hurt, but in general I think we do not need to add new stuff we have in capabilities.

**Review by susnux — APPROVED** [review]

File: apps/files/lib/App.php [hunk 2]
```diff
@@ -44,9 +45,9 @@ public static function getNavigationManager(): INavigationManager {
 	public static function extendJsConfig($settings): void {
 		$appConfig = json_decode($settings['array']['oc_appconfig'], true);
 
-		$maxChunkSize = (int)Server::get(IConfig::class)->getAppValue('files', 'max_chunk_size', (string)(100 * 1024 * 1024));
```
Comment (skjnldsv) [inline apps/files/lib/App.php:47]: We need a migration. This app config is used by many admins

**Review by skjnldsv — DISMISSED** [review]

File: apps/files/lib/App.php (hunk 2 above)
Comment (provokateurin) [reply apps/files/lib/App.php:47]: Is it there is no documentation about it and also no frontend to change it.

Comment (susnux) [reply apps/files/lib/App.php:47]: But many users use it and it was often suggested to use in our forums.
So safest would be to add a Migration file where you just transfer the value from app config to system config and delete the app config value afterwards.

Comment (provokateurin) [reply apps/files/lib/App.php:47]: Sure, I'll add that :+1:

File: apps/files/lib/Migration/ChunkedUploadConfigRepairStep.php
Comment (susnux) [inline apps/files/lib/Migration/ChunkedUploadConfigRepairStep.php:1]: Is this needed as a repair step? I think this only needed one, no?
So a normal migration would be the easiest as it automatically only runs once.

**Review by yemkareems — APPROVED** [review]

**Review by susnux — APPROVED** [review]

**Pull request merged**